src/Parser/DeathKnight/Unholy/Modules/Items/ColdHeartEfficiency.js

```diff
@@ -20,63 +20,117 @@ class ColdHeartEfficiency extends Analyzer {
 
   totalColdHeartCasts = 0;
   correctColdHeartCasts = 0;
-  buffColdHeart = 0;
+  buffColdHeart = 20; //Stacks start at 20 at start of fight
+  
   unholyStrengthStart = 0;
   unholyStrengthRemaining = 0;
+  
+  concordanceStart = 0;
+  concordanceRemaining = 0;
+  
+  khazgorothStart = 0;
+  khazgorothRemaining = 0;
+  
+  timeAtMaxStacksStart = this.owner.fight.start_time;
+  timeAtMaxStacks = 0;
+  
+  unholyStrengthDuration = 15000;
+  concordanceDuration = 10000;
+  khazgorothDuration = 15000;
+  
+  coldHeartMaxStack = 20;
+  coldHeartOptimalMinStack = 14;
+	
+  remainingDurationAllowed = 4000;
+  maxDurationAtMaxStacksAllowed = 4000;
+  
+  castsTooEarly = 0;
+  castsTooLate = 0;
+
 
   on_byPlayer_applybuff(event) {
     const spellID = event.ability.guid;
     if (spellID === SPELLS.UNHOLY_STRENGTH_BUFF.id) {
       this.unholyStrengthStart = event.timestamp;
     }
+	if (spellID === SPELLS.CONCORDANCE_OF_THE_LEGIONFALL_STRENGTH.id) {
+		this.concordanceStart = event.timestamp;
+	}
+	if (spellID === SPELLS.KHAZGOROTHS_SHAPING.id) {
+		this.khazgorothStart = event.timestamp;
+	}
   }
-
+  
+  on_byPlayer_refreshbuff(event) {
+	const spellID = event.ability.guid;
+	if (spellID === SPELLS.UNHOLY_STRENGTH_BUFF.id) {
+      this.unholyStrengthStart = event.timestamp;
+    }
+	if (spellID === SPELLS.CONCORDANCE_OF_THE_LEGIONFALL_STRENGTH.id) {
+		this.concordanceStart = event.timestamp;
+	}
+	if (spellID === SPELLS.KHAZGOROTHS_SHAPING.id) {
+		this.khazgorothStart = event.timestamp;
+	}
+  }
+  
   on_byPlayer_removebuff(event) {
     const spellID = event.ability.guid;
     if (spellID === SPELLS.UNHOLY_STRENGTH_BUFF.id) {
       this.unholyStrengthStart = 0;
     }
+	if (spellID === SPELLS.CONCORDANCE_OF_THE_LEGIONFALL_STRENGTH.id) {
+		this.concordanceStart = 0;
+    }
+	if (spellID === SPELLS.KHAZGOROTHS_SHAPING.id) {
+		this.khazgorothStart = 0;
+	}
   }
 
   on_byPlayer_applybuffstack(event) {
     const spellID = event.ability.guid;
     if (spellID === SPELLS.COLD_HEART_BUFF.id) {
       this.buffColdHeart = event.stack;
+	  if (this.buffColdHeart == 20) {
```

Comment:
This is causing the build to fail, double equals isn't allowed by our build rules, must be triple equals.